Subject: [PATCH] Updates TensorFlow Bazel version to the newly released version 5.0.0.
Migrate tensorflow from using Bazel's legacy built-in pkg_tar to using @rules_pkg.
---
Index: .bazelrc
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.bazelrc b/.bazelrc
--- a/.bazelrc	(revision a4dfb8d1a71385bd6d122e4f27f86dcebb96712d)
+++ b/.bazelrc	(revision aef23e59bf432878b5fab6797c7fe940449d9584)
@@ -128,11 +128,6 @@
 # archives in -whole_archive -no_whole_archive.
 build --noincompatible_remove_legacy_whole_archive
 
-# These are bazel 2.0's incompatible flags. Tensorflow needs to use bazel 2.0.0
-# to use cc_shared_library, as part of the Tensorflow Build Improvements RFC:
-# https://github.com/tensorflow/community/pull/179
-build --noincompatible_prohibit_aapt1
-
 build --enable_platform_specific_config
 
 # Enable XLA support by default.
@@ -482,12 +477,12 @@
 
 # Linux CPU
 build:rbe_linux_py2 --config=rbe_linux
-build:rbe_linux_py2 --repo_env=PYTHON_BIN_PATH="/usr/bin/python2"
-build:rbe_linux_py2 --python_path="/usr/bin/python2"
+build:rbe_linux_py2 --repo_env=PYTHON_BIN_PATH="/usr/bin/python"
+build:rbe_linux_py2 --python_path="/usr/bin/python"
 build:rbe_linux_py2 --repo_env=TF_PYTHON_CONFIG_REPO="@org_tensorflow//third_party/toolchains/preconfig/ubuntu16.04/py"
 
 build:rbe_linux_py3 --config=rbe_linux
-build:rbe_linux_py3 --python_path="/usr/bin/python3"
+build:rbe_linux_py3 --python_path="/usr/bin/python"
 build:rbe_linux_py3 --repo_env=TF_PYTHON_CONFIG_REPO="@ubuntu16.04-manylinux2010-py3_config_python"
 
 build:rbe_win --config=rbe
Index: .bazelversion
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.bazelversion b/.bazelversion
--- a/.bazelversion	(revision a4dfb8d1a71385bd6d122e4f27f86dcebb96712d)
+++ b/.bazelversion	(revision aef23e59bf432878b5fab6797c7fe940449d9584)
@@ -1,1 +1,1 @@
-3.7.2
+5.1.1
Index: configure.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/configure.py b/configure.py
--- a/configure.py	(revision a4dfb8d1a71385bd6d122e4f27f86dcebb96712d)
+++ b/configure.py	(revision aef23e59bf432878b5fab6797c7fe940449d9584)
@@ -47,8 +47,8 @@
 _TF_WORKSPACE_ROOT = ''
 _TF_BAZELRC = ''
 _TF_CURRENT_BAZEL_VERSION = None
-_TF_MIN_BAZEL_VERSION = '3.7.2'
-_TF_MAX_BAZEL_VERSION = '3.99.0'
+_TF_MIN_BAZEL_VERSION = '5.1.1'
+_TF_MAX_BAZEL_VERSION = '5.99.0'
 
 NCCL_LIB_PATHS = [
     'lib64/', 'lib/powerpc64le-linux-gnu/', 'lib/x86_64-linux-gnu/', ''
Index: tensorflow/tools/lib_package/BUILD
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tensorflow/tools/lib_package/BUILD b/tensorflow/tools/lib_package/BUILD
--- a/tensorflow/tools/lib_package/BUILD	(revision a4dfb8d1a71385bd6d122e4f27f86dcebb96712d)
+++ b/tensorflow/tools/lib_package/BUILD	(revision aef23e59bf432878b5fab6797c7fe940449d9584)
@@ -1,7 +1,7 @@
 # Packaging for TensorFlow artifacts other than the Python API (pip whl).
 # This includes the C API, Java API, and protocol buffer files.
 
-load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar")
+load("@rules_pkg//pkg:tar.bzl", "pkg_tar")
 load("@local_config_cuda//cuda:build_defs.bzl", "if_cuda")
 load("@local_config_syslibs//:build_defs.bzl", "if_not_system_lib")
 load("//tensorflow:tensorflow.bzl", "VERSION", "VERSION_MAJOR", "if_macos")
Index: tensorflow/workspace1.bzl
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tensorflow/workspace1.bzl b/tensorflow/workspace1.bzl
--- a/tensorflow/workspace1.bzl	(revision a4dfb8d1a71385bd6d122e4f27f86dcebb96712d)
+++ b/tensorflow/workspace1.bzl	(revision aef23e59bf432878b5fab6797c7fe940449d9584)
@@ -1,15 +1,26 @@
 """TensorFlow workspace initialization. Consult the WORKSPACE on how to use it."""
 
 load("//third_party/android:android_configure.bzl", "android_configure")
-load("//third_party/toolchains/preconfig/generate:archives.bzl", "bazel_toolchains_archive")
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 load("@com_github_grpc_grpc//bazel:grpc_deps.bzl", "grpc_deps")
 load("@io_bazel_rules_closure//closure:defs.bzl", "closure_repositories")
+load("@rules_pkg//:deps.bzl", "rules_pkg_dependencies")
 
 def workspace():
     native.register_toolchains("@local_config_python//:py_toolchain")
+    rules_pkg_dependencies()
 
     closure_repositories()
-    bazel_toolchains_archive()
+
+    http_archive(
+        name = "bazel_toolchains",
+        sha256 = "540cc8fec2bf8ab64d16fb9a7018f25738a4a03434057ea01b5d34add446ffb1",
+        strip_prefix = "bazel-toolchains-ea243d43269df23de03a797cff2347e1fc3d02bb",
+        urls = [
+            "http://mirror.tensorflow.org/github.com/bazelbuild/bazel-toolchains/archive/ea243d43269df23de03a797cff2347e1fc3d02bb.tar.gz",
+            "https://github.com/bazelbuild/bazel-toolchains/archive/ea243d43269df23de03a797cff2347e1fc3d02bb.tar.gz",
+        ],
+    )
 
     android_configure(name = "local_config_android")
 
Index: tensorflow/workspace3.bzl
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tensorflow/workspace3.bzl b/tensorflow/workspace3.bzl
--- a/tensorflow/workspace3.bzl	(revision a4dfb8d1a71385bd6d122e4f27f86dcebb96712d)
+++ b/tensorflow/workspace3.bzl	(revision aef23e59bf432878b5fab6797c7fe940449d9584)
@@ -23,6 +23,15 @@
         ],
     )
 
+    http_archive(
+        name = "rules_pkg",
+        urls = [
+            "https://mirror.bazel.build/github.com/bazelbuild/rules_pkg/releases/download/0.7.0/rules_pkg-0.7.0.tar.gz",
+            "https://github.com/bazelbuild/rules_pkg/releases/download/0.7.0/rules_pkg-0.7.0.tar.gz",
+        ],
+        sha256 = "8a298e832762eda1830597d64fe7db58178aa84cd5926d76d5b744d6558941c2",
+    )
+
 # Alias so it can be loaded without assigning to a different symbol to prevent
 # shadowing previous loads and trigger a buildifier warning.
 tf_workspace3 = workspace
Index: tensorflow/tools/ci_build/install/install_bazel.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tensorflow/tools/ci_build/install/install_bazel.sh b/tensorflow/tools/ci_build/install/install_bazel.sh
--- a/tensorflow/tools/ci_build/install/install_bazel.sh	(revision aef23e59bf432878b5fab6797c7fe940449d9584)
+++ b/tensorflow/tools/ci_build/install/install_bazel.sh	(revision a183734ce3bcc868d038c72bd22d97889cf07504)
@@ -15,7 +15,7 @@
 # ==============================================================================
 
 # Select bazel version.
-BAZEL_VERSION="3.7.2"
+BAZEL_VERSION="5.1.1"
 
 set +e
 local_bazel_ver=$(bazel version 2>&1 | grep -i label | awk '{print $3}')
Index: tensorflow/tools/ci_build/install/install_bazel_from_source.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tensorflow/tools/ci_build/install/install_bazel_from_source.sh b/tensorflow/tools/ci_build/install/install_bazel_from_source.sh
--- a/tensorflow/tools/ci_build/install/install_bazel_from_source.sh	(revision aef23e59bf432878b5fab6797c7fe940449d9584)
+++ b/tensorflow/tools/ci_build/install/install_bazel_from_source.sh	(revision a183734ce3bcc868d038c72bd22d97889cf07504)
@@ -18,7 +18,7 @@
 # It will compile bazel from source and install it in /usr/local/bin
 
 # Select bazel version.
-BAZEL_VERSION="3.7.2"
+BAZEL_VERSION="5.1.1"
 
 set +e
 local_bazel_ver=$(bazel version 2>&1 | grep -i label | awk '{print $3}')
Index: tensorflow/tools/ci_build/release/common.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tensorflow/tools/ci_build/release/common.sh b/tensorflow/tools/ci_build/release/common.sh
--- a/tensorflow/tools/ci_build/release/common.sh	(revision aef23e59bf432878b5fab6797c7fe940449d9584)
+++ b/tensorflow/tools/ci_build/release/common.sh	(revision a183734ce3bcc868d038c72bd22d97889cf07504)
@@ -17,7 +17,7 @@
 
 # Keep in sync with tensorflow_estimator and configure.py.
 # LINT.IfChange
-LATEST_BAZEL_VERSION=3.7.2
+LATEST_BAZEL_VERSION=5.1.1
 # LINT.ThenChange(
 #   //tensorflow/opensource_only/configure.py,
 #   //tensorflow_estimator/google/kokoro/common.sh,
@@ -67,7 +67,7 @@
   esac
   mkdir -p "$HOME/bin"
   wget --no-verbose -O "$HOME/bin/bazel" \
-      "https://github.com/bazelbuild/bazelisk/releases/download/v1.3.0/$name"
+      "https://github.com/bazelbuild/bazelisk/releases/download/v1.11.0/$name"
   chmod u+x "$HOME/bin/bazel"
   if [[ ! ":$PATH:" =~ :"$HOME"/bin/?: ]]; then
     PATH="$HOME/bin:$PATH"
Index: tensorflow/tools/ci_build/release/common_win.bat
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tensorflow/tools/ci_build/release/common_win.bat b/tensorflow/tools/ci_build/release/common_win.bat
--- a/tensorflow/tools/ci_build/release/common_win.bat	(revision aef23e59bf432878b5fab6797c7fe940449d9584)
+++ b/tensorflow/tools/ci_build/release/common_win.bat	(revision a183734ce3bcc868d038c72bd22d97889cf07504)
@@ -75,7 +75,7 @@
 @REM Setup Bazel
 @REM
 :: Download Bazel from github and make sure its found in PATH.
-SET BAZEL_VERSION=3.7.2
+SET BAZEL_VERSION=5.1.1
 md C:\tools\bazel\
 wget -q https://github.com/bazelbuild/bazel/releases/download/%BAZEL_VERSION%/bazel-%BAZEL_VERSION%-windows-x86_64.exe -O C:/tools/bazel/bazel.exe
 SET PATH=C:\tools\bazel;%PATH%
Index: tensorflow/workspace2.bzl
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tensorflow/workspace2.bzl b/tensorflow/workspace2.bzl
--- a/tensorflow/workspace2.bzl	(revision aef23e59bf432878b5fab6797c7fe940449d9584)
+++ b/tensorflow/workspace2.bzl	(revision a183734ce3bcc868d038c72bd22d97889cf07504)
@@ -993,34 +993,31 @@
     # https://github.com/bazelbuild/rules_apple/releases
     tf_http_archive(
         name = "build_bazel_rules_apple",
-        sha256 = "ee9e6073aeb5a65c100cb9c44b0017c937706a4ae03176e14a7e78620a198079",
-        strip_prefix = "rules_apple-5131f3d46794bf227d296c82f30c2499c9de3c5b",
+        sha256 = "a5f00fd89eff67291f6cd3efdc8fad30f4727e6ebb90718f3f05bbf3c3dd5ed7",
         urls = [
-            "https://storage.googleapis.com/mirror.tensorflow.org/github.com/bazelbuild/rules_apple/archive/5131f3d46794bf227d296c82f30c2499c9de3c5b.tar.gz",
-            "https://github.com/bazelbuild/rules_apple/archive/5131f3d46794bf227d296c82f30c2499c9de3c5b.tar.gz",
+            "https://storage.googleapis.com/mirror.tensorflow.org/github.com/bazelbuild/rules_apple/releases/download/0.33.0/rules_apple.0.33.0.tar.gz",
+            "https://github.com/bazelbuild/rules_apple/releases/download/0.33.0/rules_apple.0.33.0.tar.gz",
         ],
     )
 
     # https://github.com/bazelbuild/rules_swift/releases
     tf_http_archive(
         name = "build_bazel_rules_swift",
-        sha256 = "d0833bc6dad817a367936a5f902a0c11318160b5e80a20ece35fb85a5675c886",
-        strip_prefix = "rules_swift-3eeeb53cebda55b349d64c9fc144e18c5f7c0eb8",
+        sha256 = "8a49da750560b185804a4bc95c82d3f9cc4c2caf788960b0e21945759155fdd9",
         urls = [
-            "https://storage.googleapis.com/mirror.tensorflow.org/github.com/bazelbuild/rules_swift/archive/3eeeb53cebda55b349d64c9fc144e18c5f7c0eb8.tar.gz",
-            "https://github.com/bazelbuild/rules_swift/archive/3eeeb53cebda55b349d64c9fc144e18c5f7c0eb8.tar.gz",
+            "https://storage.googleapis.com/mirror.tensorflow.org/github.com/bazelbuild/rules_swift/releases/download/0.25.0/rules_swift.0.25.0.tar.gz",
+            "https://github.com/bazelbuild/rules_swift/releases/download/0.25.0/rules_swift.0.25.0.tar.gz",
         ],
     )
 
     # https://github.com/bazelbuild/apple_support/releases
     tf_http_archive(
         name = "build_bazel_apple_support",
-        sha256 = "ad8ae80e93612b8151019367a3d1604d7a51c14480dae1254e10252007e8260c",
-        strip_prefix = "apple_support-501b4afb27745c4813a88ffa28acd901408014e4",
+        sha256 = "c604b75865c07f45828c7fffd5fdbff81415a9e84a68f71a9c1d8e3b2694e547",
         urls = [
-            "https://storage.googleapis.com/mirror.tensorflow.org/github.com/bazelbuild/apple_support/archive/501b4afb27745c4813a88ffa28acd901408014e4.tar.gz",
-            "https://github.com/bazelbuild/apple_support/archive/501b4afb27745c4813a88ffa28acd901408014e4.tar.gz",
-        ],
+            "https://storage.googleapis.com/mirror.tensorflow.org/github.com/bazelbuild/apple_support/releases/download/0.12.1/apple_support.0.12.1.tar.gz",
+            "https://github.com/bazelbuild/apple_support/releases/download/0.12.1/apple_support.0.12.1.tar.gz",
+        ]
     )
 
     # https://github.com/bazelbuild/bazel-skylib/releases
